src/NHibernate/Transaction/AdoNetWithSystemTransactionFactory.cs

```diff
@@ -0,0 +1,528 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Transactions;
+using NHibernate.AdoNet;
+using NHibernate.Engine;
+using NHibernate.Engine.Transaction;
+using NHibernate.Impl;
+using NHibernate.Util;
+
+namespace NHibernate.Transaction
+{
+	/// <summary>
+	/// <see cref="ITransaction"/> factory implementation supporting system
+	/// <see cref="System.Transactions.Transaction"/>.
+	/// </summary>
+	public class AdoNetWithSystemTransactionFactory : AdoNetTransactionFactory
```

Comment:
I have changed the factories for having their methods overridable, thus allowing extending `AdoNetTransactionFactory` instead of duplicating it or delegating to an internal instance some behaviors.

I have also changed the system transaction factory for allowing it to be more easily extended by users for customization.